src/coreclr/jit/simd.h

```diff
@@ -841,6 +971,164 @@ TBase EvaluateBinaryScalar(genTreeOps oper, TBase arg0, TBase arg1)
     }
 }
 
+#if defined(FEATURE_MASKED_HW_INTRINSICS)
+template <typename TBase>
+void EvaluateBinaryMask(
+    genTreeOps oper, bool scalar, unsigned simdSize, simdmask_t* result, const simdmask_t& arg0, const simdmask_t& arg1)
+{
+    uint32_t count = simdSize / sizeof(TBase);
+
+#if defined(TARGET_XARCH)
+    // For xarch we have count sequential bits, but an 8 count minimum
+
+    if (count < 8)
+    {
+        count = 8;
+    }
+    assert((count == 8) || (count == 16) || (count == 32) || (count == 64));
+
+    uint64_t bitMask = static_cast<uint64_t>((static_cast<int64_t>(1) << count) - 1);
+#elif defined(TARGET_ARM64)
+    // For Arm64 we have count total bits to write, but they are sizeof(TBase) bits apart
+    uint64_t bitMask;
+
+    switch (sizeof(TBase))
+    {
+        case 1:
+        {
+            bitMask = 0xFFFFFFFFFFFFFFFF;
+            break;
+        }
+
+        case 2:
+        {
+            bitMask = 0x5555555555555555;
```

Comment:
does any of this will have to change when we add "variable length vector" support?

Reply:
When `TYP_SIMD` is added we'll likely need to consider how we plan on supporting the up to 2048-bit vectors and up to 256-bit masks.

My guess is that we'll probably just say that we don't support over 512-bit SVE for simplicity, at least until physical hardware with 1024 or 2048 bits exists. When such hardware does exist, then we'd need to expand this to handle the additional bits.